### PR TITLE
fix: Add missing OpTypes

### LINKED ISSUE #30: Add missing ops
There are multiple operations missing (`CnY` , `CnZ`, `CSdg`, `PauliExpBox`, `ConjugationBox`, ...)

Compare https://github.com/CQCL/tket-json-rs/blob/main/src/optype.rs with https://github.com/CQCL/tket/blob/develop/tket/include/tket/OpType/OpType.hpp for the full list.

### PR DESCRIPTION
**Pull request by aborgna-q**
Adds the missing optypes listed in #36.

Fixes #30.

### CODE REVIEW

File: src/opbox.rs [hunk 1]
```diff
@@ -81,6 +83,26 @@ pub enum OpBox {
...
+    /// An unordered collection of Pauli exponentials that can be synthesised in
+    /// any order, causing a change in the unitary operation. Synthesis order
+    /// depends on the synthesis strategy chosen only.
+    TermSequenceBox {
+        id: BoxID,
+        /// List of Symengine expressions.
+        pauli_gadgets: Vec<(Vec<String>, String)>,
+        /// Synthesis strategy. See [`PauliSynthStrat`].
+        #[serde(default)]
+        synthesis_strategy: PauliSynthStrat,
```
Comment (cqc-alec) [inline src/opbox.rs:95]: Can we make the names match those used in TKET serialization ( https://github.com/CQCL/tket/blob/a2f6fab8a57da8787dfae94764b7c3a8e5779024/schemas/circuit_v1.json#L832 )?

File: src/optype.rs [hunk 2]
```diff
@@ -207,6 +213,12 @@ pub enum OpType {
     /// \end{array} \right] \f$
     CSXdg,
 
+    /// Controlled \ref OpType::S gate
```
Comment (cqc-alec) [inline src/optype.rs:216]: Use rust style for cross-references rather than doxygen style. (Several more examples of this below.)

Comment (aborgna-q) [reply src/optype.rs]: Fixed

File: src/opbox.rs (hunk 1 above)
Comment (aborgna-q) [reply src/opbox.rs]: I did a full pass on the schema to ensure our definition was OK.

Apparently some of the already-present box definitions where missing or incorrect, so I also fixed them.

Comment (aborgna-q) [reply src/opbox.rs]: Unfortunately #36 won't be able to catch mismatching attributes.

To validate that we'd need to add tests that construct each possible `pytket` optype and check it against the rust code :thinking:

I may add a low-priority issue for that.

File: src/opbox.rs [hunk 3]
```diff
@@ -1,9 +1,13 @@
 //! Data definition for box operations.
+//!
+//! These definitions correspond to the operations `box` attribute in the
```
Comment (cqc-alec) [inline src/opbox.rs:3]:
```suggestion
//! These definitions correspond to the operations' `box` attribute in the
```

**Review by cqc-alec — APPROVED** [review]
Thanks!

**Pull request merged**